core/src/test/java/org/apache/calcite/rex/RexBuilderTest.java

```diff
@@ -489,6 +490,34 @@ private void checkDate(RexNode node) {
     }
   }
 
+  /** Tests {@link RexBuilder#makeExactLiteral(java.math.BigDecimal)}. */
+  @Test public void testBigDecimalLiteral() {
+    final RelDataTypeFactory typeFactory =
+            new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+    final RexBuilder builder = new RexBuilder(typeFactory);
+    final RexNode literal1 = builder.makeExactLiteral(new BigDecimal("25"));
+    assertThat(((RexLiteral) literal1).getValueAs(BigDecimal.class)
```

Comment:
Would you please factor  the common logic to `assertBigDecimalLiteral(String inputText, String expectedOutputText)`?

That would make test much more readable (it won't include repeated `builder.makeExactLiteral...`), and the failure message would be much easier to understand.

Thanks.

Reply:
Modified.